src/ops/gas.rs

```diff
@@ -29,3 +29,30 @@ impl<S: Store> AbiCall<S> for Gas {
         Ok(None)
     }
 }
+
+pub struct GasConsumed;
+
+impl GasConsumed {
+    pub const GAS_CONSUMED_CALL_COST: u64 = 9165;
+}
+
+impl<S: Store> AbiCall<S> for GasConsumed {
+    const ARGUMENTS: &'static [ValueType] = &[];
+    const RETURN: Option<ValueType> = Some(ValueType::I64);
+
+    fn call(
+        context: &mut CallContext<S>,
+        _args: RuntimeArgs,
+    ) -> Result<Option<RuntimeValue>, VMError<S>> {
+        // Note that when this function returns, the costs that invoking this
+        // host function has are not taken into account here, they are
+        // once the host call execution finnishes. Therefore, we need to
+        // add to the result the gas_consumed fn cost.
+        // FIXME: This will not always be correct since if the `gas_consumed =
+        // ALL` the gas, this will add the extra cost of the call
+        // which can't be consumed since it's not even there.
+        Ok(Some(RuntimeValue::from(
+            context.gas_consumed() + GasConsumed::GAS_CONSUMED_CALL_COST,
+        )))
+    }
```

Comment:
We should handle the FIXME already in this PR. I think we have all the information we need. If we can access to the context's gas_meter we have the gas left, and based on that we can decide how much we need to increase the return value.